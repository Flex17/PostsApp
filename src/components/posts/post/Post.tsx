import React from 'react';
import { Col, Container, Image, Row } from 'react-bootstrap';
import logo from '../../../img/avatar.jpg';
import { NavLink } from 'react-router-dom';
import { ABOUT_USER_PAGE } from '../../../App';
import { PostsI } from '../../../store/postsReducer';
import Comments from '../comments/Comments';

const Post: React.FC<PostsI> = ({
	body,
	title,
	userId,
	id,
}: PostsI) => {

    return (
        <div>
			<Container className="border border-bottom-0 rounded-bottom-0 rounded-1 p-2">
				<Row className="d-flex">
					<NavLink to={`${ABOUT_USER_PAGE}/${userId}`} className="w-25" style={{maxWidth: '150px'}}>
						<Image roundedCircle src={logo} className="w-100" alt="Post image"/>
					</NavLink>
					<Col>
						<div className="fs-3">{title}</div>
						<div className="fs-5">{body}</div>
					</Col>
				</Row>
			</Container>
			<Comments postId={id} />
		</div>
    );
};

export default Post;
