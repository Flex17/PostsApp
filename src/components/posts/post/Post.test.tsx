import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Post from './Post';

const fakePost = {
    id: 1,
    userId: 1,
    title: 'Test Title',
    body: 'Test Body',
};

const renderPost = () => render(
    <Router>
        <Post {...fakePost} />
    </Router>
);

describe('Post', () => {
    test('Рендер заголовка и текста', () => {
        renderPost();

        const titleElement = screen.getByText('Test Title');
        const bodyElement = screen.getByText('Test Body');

        expect(titleElement).toBeTruthy();
        expect(bodyElement).toBeTruthy();
    });

    test('Рендер изображения', () => {
        renderPost();

        const imageElement = screen.getByAltText('Post image');

        expect(imageElement).toBeTruthy();
        expect(imageElement.getAttribute('src')).toContain('avatar.jpg');
    });

    test('Posts snapshot', () => {
        const page = renderPost();

        expect(page).toMatchSnapshot();
    });
});
