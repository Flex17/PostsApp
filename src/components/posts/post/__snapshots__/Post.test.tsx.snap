// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Posts snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="border border-bottom-0 rounded-bottom-0 rounded-1 p-2 container"
        >
          <div
            class="d-flex row"
          >
            <a
              class="w-25"
              href="/about_user/1"
              style="max-width: 150px;"
            >
              <img
                alt="Post image"
                class="w-100 rounded-circle"
                src="avatar.jpg"
              />
            </a>
            <div
              class="col"
            >
              <div
                class="fs-3"
              >
                Test Title
              </div>
              <div
                class="fs-5"
              >
                Test Body
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="accordion mb-4"
            id="accordionExample1"
          >
            <div
              class="accordion-item"
            >
              <h2
                class="accordion-header"
                id="headingTwo1"
              >
                <button
                  aria-controls="collapseTwo1"
                  aria-expanded="false"
                  class="accordion-button collapsed"
                  data-bs-target="#collapseTwo1"
                  data-bs-toggle="collapse"
                  data-testid="btn"
                  type="button"
                >
                  Посмотреть комментарии
                </button>
              </h2>
              <div
                aria-labelledby="headingTwo1"
                class="accordion-collapse collapse"
                data-bs-parent="#accordionExample1"
                id="collapseTwo1"
              >
                <div
                  class="accordion-body"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="border border-bottom-0 rounded-bottom-0 rounded-1 p-2 container"
      >
        <div
          class="d-flex row"
        >
          <a
            class="w-25"
            href="/about_user/1"
            style="max-width: 150px;"
          >
            <img
              alt="Post image"
              class="w-100 rounded-circle"
              src="avatar.jpg"
            />
          </a>
          <div
            class="col"
          >
            <div
              class="fs-3"
            >
              Test Title
            </div>
            <div
              class="fs-5"
            >
              Test Body
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="accordion mb-4"
          id="accordionExample1"
        >
          <div
            class="accordion-item"
          >
            <h2
              class="accordion-header"
              id="headingTwo1"
            >
              <button
                aria-controls="collapseTwo1"
                aria-expanded="false"
                class="accordion-button collapsed"
                data-bs-target="#collapseTwo1"
                data-bs-toggle="collapse"
                data-testid="btn"
                type="button"
              >
                Посмотреть комментарии
              </button>
            </h2>
            <div
              aria-labelledby="headingTwo1"
              class="accordion-collapse collapse"
              data-bs-parent="#accordionExample1"
              id="collapseTwo1"
            >
              <div
                class="accordion-body"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
